### ISSUE TITLE
Initializing workgroup memory

### ISSUE THREAD

**Issue by ether42** [EmbarkStudios/rust-gpu#1077]
As far as I can tell, the workgroup/shared memory ([context](https://github.com/EmbarkStudios/rust-gpu/issues/695)) can not be declared as to be zero-initialized and one has to do it manually.

https://github.com/KhronosGroup/Vulkan-Docs/issues/1457 has a bunch of interesting links: the extension [`VK_KHR_zero_initialize_workgroup_memory`](https://github.com/KhronosGroup/Vulkan-Guide/blob/main/chapters/extensions/shader_features.adoc#vk_khr_zero_initialize_workgroup_memory) (promoted to core in Vulkan 1.3) allows to declare workgroup/shared variables with a [null initializer](https://github.com/KhronosGroup/VK-GL-CTS/blob/1d91116914824bc86bbf4d1a597be1bbb9e495c7/external/vulkancts/modules/vulkan/compute/vktComputeZeroInitializeWorkgroupMemoryTests.cpp#L232). On the SPIR-V side, [`OpVariable`](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpVariable) with a `Workgroup` storage class is allowed to be initialized with [`OpConstantNull`](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpConstantNull) only ([like so](https://github.com/alan-baker/SPIRV-Tools/blob/126c518aa78fc1972eed30f09bfed6f77b74c2d2/test/val/val_memory_test.cpp#L4120-L4121)).

I don't know what the project would prefer, I could be an option to pass to the `SpirvBuilder` (I assume it could be desirable to always initialize these variables) or another symbol (similarly to how `storage_buffer` allows extra arguments): 
```rust
#[spirv(compute(threads(256)))]  
fn compute(
    #[spirv(workgroup, null_initialized = true)] array: &mut [u32; N],
    // ...
) {
  // ...
}
```

**Comment by Patryk27** [comment]
Btw, [since recently](https://github.com/gfx-rs/naga/pull/2111), Naga automatically includes some glue code that zero-initializes workgroup memory

**Comment by ether42** [comment]
Yeah, but unless I'm handling it wrong, it doesn't support a bunch of features and that limit its usefulness (e.g.: subgroups or atomics).
